core/src/main/java/org/mule/runtime/core/internal/management/stats/PayloadStatisticsCollection.java

```diff
@@ -44,4 +58,15 @@ public Spliterator spliterator() {
     populator.accept(size());
     return super.toArray(object);
   }
+
+  @Override
+  public void writeExternal(ObjectOutput out) throws IOException {
+    out.writeObject(getCollection());
+
+  }
+
+  @Override
+  public void readExternal(ObjectInput in) throws IOException, ClassNotFoundException {
+    this.collection = (Collection) in.readObject();
```

Comment:
we shouldn't be expecting to encounter this actual class serialized.

Reply:
Needed for the case where we are not using kryo.